rotate.go

```diff
@@ -364,19 +365,81 @@ func (r *Rotator) SetSecret(ctx context.Context, event map[string]string) error
 	}
 	curUsername, curPassword := r.ss.Credentials(curVals)
 
+	curCred := db.Credentials{
+		Username: curUsername,
+		Password: curPassword,
+	}
+
+	newCred := db.Credentials{
+		Username: newUsername,
+		Password: newPassword,
+	}
+
 	// Combine the current and new credentials. This is plumbed all the way down
 	// into the db.PassswordSetter implementation.
 	creds := db.NewPassword{
-		Current: db.Credentials{
-			Username: curUsername,
-			Password: curPassword,
-		},
-		New: db.Credentials{
-			Username: newUsername,
-			Password: newPassword,
-		},
+		Current: curCred,
+		New:     newCred,
+	}
+
+	// Check to see if DB is already set to Pending password.
+	// This can happen if there's a previous run that did not complete successfully.
```

Comment:
Let's enhance this comment to make it more specific wrt what has motivated this PR. "Did not complete successfully" can mean anything. But specifically (or as one example), we're dealing with Lambda itself crashing mid-rotate.

Reply:
Updated